sdk/azidentity/aad_identity_client.go

```diff
@@ -111,14 +143,19 @@ func (c *aadIdentityClient) authenticateCertificate(ctx context.Context, tenantI
 		return c.createAccessToken(resp)
 	}
 
-	return nil, &AuthenticationFailedError{inner: newAuthenticationResponseError(resp)}
+	return nil, &AuthenticationFailedError{inner: newAADAuthenticationFailedError(resp)}
 }
 
-func (c *aadIdentityClient) createAccessToken(res *azcore.Response) (*azcore.AccessToken, error) {
-	value := &internalAccessToken{}
+func (c *aadIdentityClient) createAccessToken(res *azcore.Response) (*tokenResponse, error) {
+	value := struct {
+		Token        string      `json:"access_token"`
+		RefreshToken string      `json:"refresh_token"`
+		ExpiresIn    json.Number `json:"expires_in"`
+		ExpiresOn    string      `json:"expires_on"`
+	}{}
 	accessToken := &azcore.AccessToken{}
 	if err := json.Unmarshal(res.Payload, &value); err != nil {
-		return nil, fmt.Errorf("internalAccessToken: %w", err)
+		return nil, fmt.Errorf("internal AccessToken: %w", err)
 	}
 	t, err := value.ExpiresIn.Int64()
 	if err != nil {
```

Comment:
I would wrap this error with some context so we know what failed (similar to the above wrapping).

Reply:
I'm not sure I understood what you mean. You want me to wrap this unmarshal error in an error type??

Reply:
On line 144 we wrap the error from `json.Unmarshal()` with an "internal AccessToken" message.  I think we should do the same thing here.